lua/nvim-tree/actions/tree/resize.lua

```diff
@@ -0,0 +1,51 @@
+local view = require "nvim-tree.view"
+
+local M = {}
+
+---Resize the tree, persisting the new size.
+---@param opts ApiTreeResizeOpts|nil
+function M.fn(opts)
+  if opts == nil then
+    -- reset to config values
+    view.configure_width()
+    view.resize()
+    return
+  end
+
+  local options = opts or {}
+  local width_cfg = options.width
+
+  if width_cfg ~= nil then
+    view.configure_width(width_cfg)
+    view.resize()
+    return
+  end
+
+  if not view.is_width_determined() then
```

Comment:
Rather than using this function, could we simply test `type(M.config.view.width) == "function"` ?

More encapsulation...

Reply:
@alex-courtis 
Hmmmmmmmm

In this case we wanna use 
```lua
M.config.view.configure_width(...)
M.config.view.resize()
```
as well (because we anyway using it)

In this case we can get rid of 
```lua
local view = require "nvim-tree.view"
```
and just do 
```lua
local view = M.config.view
```
which actually won't work...

IMO let's stay consistent (current solution). But if you OK with having `local view = require(...)` and `M.config.view` in the same function - I don't mind changing. it :)

Reply:
Fair enough, thanks for digging.

That change would indeed "taint" resize; view should be fixed but not today.